### PR TITLE
Add metadata to STATS hdf5 file #24

### PR DESCRIPTION
**Pull request by nepstad**
- PySilcam version and timestamp written to hdf5 STATS file
- Restructure and clean up STATS file initialization
- Begin refactor to replace PYSILCAM_OFFSET environment variable

### CODE REVIEW

File: pysilcam/__main__.py [hunk 1]
```diff
@@ -657,47 +675,46 @@ def configure_logger(settings):
...
+    # subtract the number of background images, so we get data from the correct start point
+    offset -= settings.Background.num_images
+    # and check offset is still positive
+    if offset < 0:
+        offset = 0
+    offset = str(offset)
+    os.environ['PYSILCAM_OFFSET'] = offset
+    logger.info('PYSILCAM_OFFSET set to: ' + offset)
+    print('PYSILCAM_OFFSET set to: ' + offset)
+
```
Comment (tankefugl) [inline pysilcam/__main__.py:719]: W293: blank line contains whitespace

File: pysilcam/__main__.py [hunk 2]
```diff
@@ -657,47 +675,46 @@ def configure_logger(settings):
...
+    print('  OK.')
+    last_time = pd.to_datetime(oldstats['timestamp'].max())
+
+    logger.info('Calculating spooling offset')
+    print('Calculating spooling offset')
+
+    # TODO: move this import to the top
+    from pysilcam.fakepymba import silcam_name2time
+    files = [f for f in sorted(os.listdir(datapath))
+                if f.endswith('.silc' or f.endswith('.bmp'))]
```
Comment (tankefugl) [inline pysilcam/__main__.py:704]: E127: continuation line over-indented for visual indent

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/__main__.py [hunk 3]
```diff
@@ -657,47 +675,45 @@ def configure_logger(settings):
...
+    offset -= settings.Background.num_images
+    # and check offset is still positive
+    if offset < 0:
+        offset = 0
+    offset = str(offset)
+    os.environ['PYSILCAM_OFFSET'] = offset
+    logger.info('PYSILCAM_OFFSET set to: ' + offset)
+    print('PYSILCAM_OFFSET set to: ' + offset)
+    
+    return offset
```
Comment (tankefugl) [inline pysilcam/__main__.py:719]: W292: no newline at end of file

File: pysilcam/__main__.py [hunk 4]
```diff
@@ -657,47 +675,45 @@ def configure_logger(settings):
...
+    # subtract the number of background images, so we get data from the correct start point
+    offset -= settings.Background.num_images
+    # and check offset is still positive
+    if offset < 0:
+        offset = 0
+    offset = str(offset)
+    os.environ['PYSILCAM_OFFSET'] = offset
+    logger.info('PYSILCAM_OFFSET set to: ' + offset)
+    print('PYSILCAM_OFFSET set to: ' + offset)
+
```
Comment (tankefugl) [inline pysilcam/__main__.py:718]: W293: blank line contains whitespace

**Review by tankefugl — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

**Review by emlynjdavies — DISMISSED** [review]
Logic looks good. I don't think there is good testing of overwriteSTATS. This is described in #83

File: pysilcam/__main__.py [hunk 5]
```diff
@@ -605,7 +623,7 @@ def collect_rts(settings, rts, stats_all):
         rts.to_csv(filename)
 
 
-def writeCSV(datafilename, stats_all):
+def write_stats(datafilename, stats_all):
```
Comment (emlynjdavies) [inline pysilcam/__main__.py:626]: I wonder if this function should be in process.py? I could be used externally from __main__ in some cases.

**Comment by emlynjdavies** [comment]
What happens when we use real pymba and not fakepymba (i.e. when the camera is attached?

File: pysilcam/__main__.py [hunk 6]
```diff
@@ -657,47 +656,45 @@ def configure_logger(settings):
...
+    # subtract the number of background images, so we get data from the correct start point
+    offset -= settings.Background.num_images
+    # and check offset is still positive
+    if offset < 0:
+        offset = 0
+    offset = str(offset)
+    os.environ['PYSILCAM_OFFSET'] = offset
+    logger.info('PYSILCAM_OFFSET set to: ' + offset)
+    print('PYSILCAM_OFFSET set to: ' + offset)
+
```
Comment (tankefugl) [inline pysilcam/__main__.py:699]: W293: blank line contains whitespace

**Review by tankefugl — DISMISSED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: pysilcam/__main__.py (hunk 5 above)
Comment (nepstad) [reply pysilcam/__main__.py]: Done

**Review by emlynjdavies — APPROVED** [review]

**Review by emlynjdavies — DISMISSED** [review]

**Review by emlynjdavies — APPROVED** [review]

**Review by WillRNaylor — APPROVED** [review]

**Pull request merged**